pkg/eosclient/eosgrpc/eosgrpc.go

```diff
@@ -849,76 +761,78 @@ func (c *Client) rm(ctx context.Context, username, path string) error {
 	// Now send the req and see what happens
 	resp, err := c.cl.Exec(ctx, rq)
 	if err != nil {
-		log.Warn().Err(err).Str("username", username).Str("path", path).Str("err", err.Error())
+		log.Warn().Err(err).Str("path", path).Str("err", err.Error())
 		return err
 	}
 
 	if resp == nil {
-		return errtypes.InternalError(fmt.Sprintf("nil response for username: '%s' path: '%s'", username, path))
+		return errtypes.InternalError(fmt.Sprintf("nil response for uid: '%s' path: '%s'", uid, path))
 	}
 
-	log.Info().Str("username", username).Str("path", path).Str("resp:", fmt.Sprintf("%#v", resp)).Msg("grpc response")
+	log.Info().Str("path", path).Str("resp:", fmt.Sprintf("%#v", resp)).Msg("grpc response")
 
 	return err
 
 }
 
-func (c *Client) rmdir(ctx context.Context, username, path string) error {
+func (c *Client) rmdir(ctx context.Context, uid, gid, path string) error {
 	log := appctx.GetLogger(ctx)
 
 	// Initialize the common fields of the NSReq
-	rq, err := c.initNSRequest(username)
+	rq, err := c.initNSRequest(uid, gid)
 	if err != nil {
 		return err
 	}
 
-	msg := new(erpc.NSRequest_RmdirRequest)
+	msg := new(erpc.NSRequest_RmRequest)
 
 	msg.Id = new(erpc.MDId)
 	msg.Id.Path = []byte(path)
+	msg.Recursive = true
+	msg.Norecycle = true
```

Comment:
We would want the deleted directories to be present in the recycle bin. Same for deleting files.

Reply:
Ok